### PR TITLE
Add responding to events

### PR DESCRIPTION
**Pull request by imnotmoon**
WIP
스크린샷 2022-09-13 오전 1 07 37

### CODE REVIEW

File: docs/learn/Adding Interactivity/responding-to-events.mdx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+:::danger
+React에서 `onScroll`을 제외한 모든 이벤트는 전파됩니다. `onScroll`은 이벤트를 붙인 JSX 태그에서만 동작합니다.
+:::
+
+## 전파 막기
+
+이벤트 핸들러는 **이벤트 객체**를 인자로 전달받습니다. 관습적으로 "event"를 줄여 `e`라고 부릅니다. 이 객체를 통해 이벤트에 대한 정보를 읽을 수 있습니다.
+
+또한 이벤트 객체를 통해 전파를 막는것도 가능하빈다. 만약 부모 컴포넌트에 도달하기 전에 이벤트의 전파를 막으려 한다면,
+아래 예시처럼 `e.stopPropagation()`을 호출하면 됩니다.
```
Comment (Songycs) [inline docs/learn/Adding Interactivity/responding-to-events.mdx]:
```suggestion
또한 이벤트 객체를 통해 전파를 막는것도 가능합니다. 만약 부모 컴포넌트에 도달하기 전에 이벤트의 전파를 막으려 한다면,
아래 예시처럼 `e.stopPropagation()`을 호출하면 됩니다.
```

File: docs/learn/Adding Interactivity/responding-to-events.mdx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    - `Toolbar` 컴포넌트의 prop으로 전달받은 `onClick` 함수를 호출합니다.
+3. `Toolbar`에 정의된 함수는 버튼만의 alert를 띄우도록 합니다.
+4. 전파가 멈췄기 때문에 부모인 `<div>`의 `onClick`은 실행되지 않습니다.
+
+`e.stopPropagation()` 때문에 버튼을 클릭했을때 이제는 두개(`<button>`과 부모 툴바인 `<div>`)가 아닌 한개(`<button>`)의 alert만 나타나게 됩니다.
+버튼을 클릭하는 것은 이를 둘러싼 툴바를 클릭하는 것과 동일하지 않으므로, 전파를 막는게 이 상황에서는 더 적절한 UI입니다.
+
+:::info Deep Dive
+### 이벤트의 캡쳐링(capture) 단계
+자식 요소의 이벤트들이 전파를 막았다 하더라도 자식 요소에서 발생한 모든 이벤트들을 캐치해야 하는 경우가 드물게 발생합니다. 예를 들어,
```
Comment (Songycs) [inline docs/learn/Adding Interactivity/responding-to-events.mdx]:
```suggestion
자식 요소의 이벤트들이 _전파를 막았다_ 하더라도 자식 요소에서 발생한 모든 이벤트들을 캐치해야 하는 경우가 드물게 발생합니다. 예를 들어,
```

File: docs/learn/Adding Interactivity/responding-to-events.mdx [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+아래 세 단계를 통해 유저가 버튼을 클릭한 경우 메시지를 보여주도록 할 수 있습니다.
+
+1. `Button` 컴포넌트 내부에 `handleClick` 함수를 선언합니다.
+2. 함수 내부에 로직을 구현합니다(`alert`를 이용해 메시지를 보여주도록 합니다).
+3. `<button>` JSX에 `onClick={handleClick}`을 추가합니다.
+
+<Code src="https://codesandbox.io/embed/sandpack-project-1p4qyg?fontsize=14&hidenavigation=1&theme=dark" />
+
+여러분은 `handleClick` 함수를 정의해 `<button>`에
+[prop으로써 전달](https://beta.reactjs.org/learn/passing-props-to-a-component)했습니다.
```
Comment (ChipmunkForLove) [inline docs/learn/Adding Interactivity/responding-to-events.mdx]:
```suggestion
[prop으로 전달](https://beta.reactjs.org/learn/passing-props-to-a-component)했습니다.
```

File: docs/learn/Adding Interactivity/responding-to-events.mdx [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+**`handleClick`은 이벤트 핸들러입니다.** 이밴트 핸들러는
+- 보통 컴포넌트 _내부에_ 정의됩니다.
+- `handle` 로 시작하는 이름을 갖습니다. 그 뒤에는 이벤트 이름이 붙습니다.
+
+:::info
+관습적으로, 이벤트 핸들러의 이름은 `handle`과 그 뒤에 이벤트 이름이 붙습니다.
```
Comment (ChipmunkForLove) [inline docs/learn/Adding Interactivity/responding-to-events.mdx]:
```suggestion
관습적으로, 이벤트 핸들러의 이름은 `handle` 뒤에 이벤트 이름이 붙습니다.
```

File: docs/learn/Adding Interactivity/responding-to-events.mdx [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+
+:::danger
+이벤드 핸들러로 전달되는 함수들은 전달만 되어야 하고, 호출되어서는 안됩니다. 예를 들어
+
+| 함수를 전달(O) | 함수를 호출(X) |
+|----|----|
+| `<button onClick={handleClick}>` | `<button onClick={handleClick()}>` |
+
+둘의 차이는 미묘합니다. 첫 번째 예시에서 `handleClick` 함수는 `onClick` 이벤트 핸들러로써 전달되었습니다.
+이는 React에게 함수를 기억하고 버튼이 클릭되었을 때 그 함수를 호출하라고 명령하는 것입니다.
```
Comment (ChipmunkForLove) [inline docs/learn/Adding Interactivity/responding-to-events.mdx]:
```suggestion
이는 React에게 함수를 기억하게 하고 버튼이 클릭되었을 때 그 함수를 호출하라고 명령하는 것입니다.
```

File: docs/learn/Adding Interactivity/responding-to-events.mdx [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+
+- `<button>`과 같은 요소의 prop에 함수를 전달함으로써 이벤트를 제어할 수 있습니다.
+- 이벤트 핸들러는 반드시 전달만 되어야 하고, **호출되어서는 안됩니다!** `onClick={handleClick}`같이 전달되어야 하고 `onClick={handleClick()}`은 안됩니다.
+- 이벤트 핸들러는 인라인으로 선언될 수도 있고, JSX와 분리되어 선언될 수도 있습니다.
+- 이벤트 핸들러는 컴포넌트 안에 정의되므로 props에 접근하는 것이 가능합니다.
+- 이벤트 핸들러를 부모 컴포넌트에서 정의한 후 자식에게 prop으로 넘겨줄 수 있습니다.
+- 앱에 연관된 이름으로 이벤트 핸들러의 이름을 짓는것도 가능합니다.
+- 이벤트는 위로 전파됩니다. 핸들러의 첫번째 인자를 통해 `e.stopPropagation()`을 호출하는 것으로 이를 막을 수 있습니다.
+- 이벤트는 브라우저의 기본 동작에 의해 원하지 않는 동작을 발생시킬 수 있습니다. `e.preventDefault()`를 통해 이를 막을 수 있습니다.
+- 이벤트 핸들러를 prop으로 받아 자식의 핸들러에서 호출하는 것은 전파를 막기 위한 좋은 방법입니다.
```
Comment (Songycs) [inline docs/learn/Adding Interactivity/responding-to-events.mdx]:
```suggestion
- 이벤트 핸들러를 prop으로 받아 자식의 핸들러에서 호출하는 것은 전파의 좋은 대안입니다.
```

**Review by Songycs — APPROVED** [review]
👍 👍 👍

**Pull request merged**